echarts.volkovlabs.io/bar.json

```diff
@@ -62,18 +62,72 @@
           "height": 600
         },
         "editorMode": "code",
+        "visualEditor": {
+          "dataset": [
+            {
+              "name": "Category",
+              "source": "A"
+            },
+            {
+              "name": "Value",
+              "source": "A"
+            }
+          ],
+          "series": [
+            {
+              "uid": "f86dc46d-1364-4d2b-8d5f-834633903124",
+              "id": "A",
+              "name": "Bar",
+              "type": "bar"
+            }
+          ],
+          "code": "console.log(context);\nreturn {\n  dataset: context.editor.dataset,\n  series: context.editor.series,\n  xAxis: {\n    type: 'category',\n  },\n  yAxis: {\n    type: 'value',\n    // min: 'dataMin',\n  },\n}\n",
+          "codeHeight": 600
+        },
+        "themeEditor": {
+          "name": "default",
+          "height": 400,
+          "config": "{}"
+        },
         "gaode": {
           "key": "",
           "plugin": "AMap.Scale,AMap.ToolBar"
         },
-        "getOption": "return {\n  grid: {\n    bottom: \"3%\",\n    containLabel: true,\n    left: \"3%\",\n    right: \"4%\",\n    top: \"4%\"\n  },\n  xAxis: {\n    type: 'category',\n    data: ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun']\n  },\n  yAxis: {\n    type: 'value'\n  },\n  series: [\n    {\n      data: [120, 200, 150, 80, 70, 110, 130],\n      type: 'bar',\n      showBackground: true,\n      backgroundStyle: {\n        color: 'rgba(180, 180, 180, 0.2)'\n      }\n    }\n  ]\n};",
+        "getOption": "data.series.map((s) => {\n  categories = s.fields.find((f) => f.name === 'Category').values;\n  values = s.fields.find((f) => f.name === 'Value').values;\n});\n\nreturn {\n  grid: {\n    bottom: \"3%\",\n    containLabel: true,\n    left: \"3%\",\n    right: \"4%\",\n    top: \"4%\"\n  },\n  xAxis: {\n    type: 'category',\n    data: categories\n  },\n  yAxis: {\n    type: 'value'\n  },\n  series: [\n    {\n      data: values,\n      type: 'bar',\n      showBackground: true,\n      backgroundStyle: {\n        color: 'rgba(180, 180, 180, 0.2)'\n      }\n    }\n  ]\n};",
         "google": {
           "callback": "gmapReady",
           "key": ""
         },
         "map": "json",
         "renderer": "canvas"
       },
+      "targets": [
+        {
+          "datasource": {
+            "type": "marcusolsson-static-datasource",
+            "uid": "4ytGZmF4k"
+          },
+          "refId": "A",
+          "frame": {
+            "fields": [
+              {
+                "name": "Category",
+                "type": "string",
+                "config": {},
+                "values": ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"]
+              },
+              {
+                "name": "Value",
+                "type": "string",
```

Comment:
Please change number for all places where values are numbers

Reply:
updated